src/api/users.js

```diff
@@ -0,0 +1,64 @@
+const express = require("express");
+const usersExample = require("./dev/usersExample");
+
+const router = express.Router();
+
+router.get("/", (req, res) => {
+  res.status(200).json(usersExample);
+});
+
+router.get("/:id", (req, res) => {
+  const result = usersExample.find((user) => user.id === +req.params.id);
+  if (result) res.status(200).json(result);
+  else
+    res.status(404).json({
+      message: "No user found",
+    });
+});
+
+router.post("/", (req, res) => {
+  const initial = usersExample;
+  const newUser = {
+    id: [...initial].pop().id + 1,
+    ...req.body,
+  };
+  initial.push(newUser);
+  res.status(201).json(initial[initial.length - 1]);
+});
+
+router.put("/:id", (req, res) => {
+  const id = +req.params.id;
+  const index = usersExample.indexOf(
+    usersExample.find((user) => user.id === id)
+  );
+  if (index >= 0) {
+    let elementToUpdate = usersExample.find((user) => user.id === id);
+    elementToUpdate = {
+      id,
+      ...req.body,
```

Comment:
If the body contain only modified fields, original datas will be deleted (have to do like companies path)